classes/components/forms/context/PKPAppearanceSetupForm.inc.php

```diff
@@ -42,14 +42,33 @@ public function __construct($action, $locales, $context, $baseUrl, $temporaryFil
 		$this->locales = $locales;
 
 		$sidebarOptions = [];
+		$enabledOptions = [];
+		$disabledOptions = [];
+
+		$currentBlocks = (array) $context->getData('sidebar');
+
 		$plugins = \PluginRegistry::loadCategory('blocks', true);
+
+		foreach ($currentBlocks as $plugin) {
+			if ($plugins[$plugin]) {
```

Comment:
Sorry to be picky, but this will throw a PHP Notice in newer versions of PHP, when the array key is not set. I think you need to do:

```php
if (isset($plugins[$plugin])) {
````